HHS/Head-Start-TTADP
# [TTAHUB-1085] Update form state when saving draft on goals and objectives page

## Description of change
This PR fixes the bugs below.

## How to test

**Data lingers in the local storage. Some data stays in the local storage and gives the users an incorrect view of what is in the database (Krys)**

Steps to see one case:

    create a new AR (I tried with 2 recipients)

    select an existing goal and click "Save Draft"
    navigate away from the page through the navigator
    come back and remove the goal
    select a different existing goal and use "Save Draft"
    reload the page
    observe that the old removed goal shows instead of the new one.

**Newly refreshed goal disappears when a page is refreshed. (Krys - this might be related to the above)**

Steps:

    create an AR (I used 2 different recipients)
    select an existing goal and click "Save Draft"
    refresh the page to see the goal disappear

**Navigating away from the "Goals and Objectives" section on an AR doesn't save data. (Krys)**

Steps:

    create a goal on a new AR
    without pressing "Save Draft", navigate away by selecting a different section, e.g. "Next Steps"
    come back to the "Goals & Objectives". 
    note no goals show

**Editing a newly imported goal on the RTR page changes the "createdVia"**

To simulate this scenario:

    find a goal that was imported from the RTTAPA that doesn't have an end date (if unable to find, change the "createdVia" and "isFromSmartsheetTtaPlan" in the db for any goal without the end date to simulate an imported goal)

    add an end date to the goal on the RTR page
    in the db, the "createdVia" will now be "rtr" and the "isFromSmartsheetTtaPlan" flag false

**Imported goals that don't have status are missing from the goal selection dropdown on an AR**

Steps:

    find a goal that was imported from the RTTAPA with a blank status
    create an activity report and try adding that goal to it
    the goal will not appear on the dropdown
    change the goal's status to "In Progress"
    the goal is now present in the dropdown on the AR dropdown.


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1085


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
